docs/mvvm/source-gen/diagnostics/MVVMTK0001.md

```diff
@@ -0,0 +1,46 @@
+---
+title: MVVM Toolkit error MVVMTK0001
+author: Sergio0694
+description: MVVM Toolkit error MVVMTK0001
+keywords: community toolkit, dotnet, csharp, mvvm, net core, net standard, source generators
+dev_langs:
+  - csharp
+---
+
+# MVVM Toolkit error MVVMTK0001
+
+Cannot apply `[INotifyPropertyChanged]` to a type that already declares the `INotifyPropertyChanged` interface.
+
+The following sample generates MVVMTK0001:
+
+```csharp
+using System.ComponentModel;
+using CommunityToolkit.Mvvm.ComponentModel;
+
+namespace MyApp;
+
+// The type already declares the interface
+[INotifyPropertyChanged]
+public partial class SampleViewModel : INotifyPropertyChanged
+{
+    public event PropertyChangedEventHandler? PropertyChanged;
+}
+```
+
+And the same goes for inherited interface implementations:
+
+```csharp
+using CommunityToolkit.Mvvm.ComponentModel;
+
+namespace MyApp;
+
+// The type inherits from ObservableObject, which implements INotifyPropertyChanged
+[INotifyPropertyChanged]
+public partial class SampleViewModel : ObservableObject
+{
+}
+```
+
+## Additional resources
+
+- You can find more examples in the [unit tests](https://github.com/CommunityToolkit/dotnet/tree/main/tests/CommunityToolkit.Mvvm.SourceGenerators.UnitTests).
```

Comment:
I think my main note is that I'm finding these links which just dump to a directory in the repo not as helpful. Could we link to the specific test case file at least?

Otherwise, I think either eventually we should link to the main MVVM Toolkit Source Generator docs and maybe that's it?

Reply:
My issue linking to the exact test is that it'd be more prone to breaking if we changed that code, as lines would be different. Or, we would have to use a permalink, which would mean people would potentially see outdated code. Here instead we're just opening the test, and people can still just open the file and look for the test for that specific error, so it seemed like a good enough compromise 🤷

Reply:
Should we just remove them for now? I'd like to figure out a way in the future to do automated link-back from source for Labs, so we can tackle this in the future?

And once we add the actual docs, we should just come back and annotate these linking to the main attribute article that they're associated with.

Reply:
I think it's fine to leave the link, it's better than nothing and it's consistent with the other docs pages 🙂
We can always improve this later on but at least people here will find something if they wanted more code to read.